Core/include/Acts/Seeding/PathSeeder.hpp

```diff
@@ -0,0 +1,250 @@
+    Vector3 ipDir;
+
+    /// The IP vertex position
+    Vector3 ipVertex;
+
+    /// The source links associated with the seed
+    std::vector<SourceLink> sourceLinks;
+
+    Seed() = delete;
+    Seed(ActsScalar ipPmag, Vector3 ipPdir, Vector3 ipPos,
+         std::vector<SourceLink> sls)
+        : ipP(ipPmag),
+          ipDir(std::move(ipPdir)),
+          ipVertex(std::move(ipPos)),
+          sourceLinks(std::move(sls)) {};
+  };
+
+  /// @brief Delegate to transform the source link to the
+  /// appropriate global frame.
+  ///
+  /// @arg The geometry context to use
+  /// @arg The source link to calibrate
+  ///
+  /// @return The global position of the source link measurement
+  using SourceLinkCalibrator =
+      Delegate<Vector3(const GeometryContext&, const SourceLink&)>;
+
+  /// @brief Delegate to provide the path width around
+  /// the intersection point to pull the source links
+  /// from the grid
+  ///
+  /// @arg The geometry context to use
+  /// @arg The geometry identifier to use if the
+  /// path width is varied across different tracking layers
+  ///
+  /// @return The path width in the bin0 and bin1 direction
+  /// defined with respect to the surface normal
+  using PathWidthLookup = Delegate<std::pair<ActsScalar, ActsScalar>(
+      const GeometryContext&, const GeometryIdentifier&)>;
+
+  /// @brief Delegate to find the intersections for the given pivot
+  /// source link
+  ///
+  /// @arg The geometry context to use
+  /// @arg The global position of the pivot source link
+  /// @arg The momentum direction of the pivot source link
+  /// at the first tracking layer
+  /// @arg The IP momentum magnitude
+  /// @arg The particle charge
+  using IntersectionLookup =
+      Delegate<std::vector<std::pair<GeometryIdentifier, Vector3>>(
+          const GeometryContext&, const Vector3&, const Vector3&,
+          const ActsScalar&, const ActsScalar&)>;
+
+  /// @brief Delegate to estimate the IP parameters
+  /// and the momentum direction at the first tracking layer
+  ///
+  /// @arg The geometry context to use
+  /// @arg The global position of the pivot source link
+  ///
+  /// @return Particle charge, the IP momentum magnitude, the IP vertex position,
+  /// the IP momentum direction, the momentum direction at the
+  /// first tracking layer
+  using TrackEstimator =
+      Delegate<std::tuple<ActsScalar, ActsScalar, Vector3, Vector3, Vector3>(
+          const GeometryContext&, const Vector3&)>;
+
+  /// @brief The nested configuration struct
+  struct Config {
+    /// Binned SourceLink provider
+    std::shared_ptr<ISourceLinkGrid<axis_t>> sourceLinkGrid;
+    /// Parameters estimator
+    TrackEstimator trackEstimator;
+    /// SourceLink calibrator
+    SourceLinkCalibrator sourceLinkCalibrator;
+    /// Intersection finder
+    IntersectionLookup intersectionFinder;
+    /// Path width provider
+    PathWidthLookup pathWidthProvider;
+    /// First layer extent
+    Extent firstLayerExtent;
```

Comment:
Couldn't that just be the first layer surface bounds object? That would avoid having to manually synchronize this object from geometry information.

Reply:
I've tried that in one of the local iterations of the code. Doesn't work well when the first surface is ambiguous, i.e. you have multiple sensors in the first tracking layer. Introduces a lot of redundant checks.